use-shopping-cart/core/stripe-middleware.js

```diff
@@ -18,11 +18,36 @@ export const handleStripe = (store) => (next) => async (action) => {
       })
     }
 
-    if (store.getState().mode === 'client-only') {
+    if (cart.cartMode === 'client-only') {
       const checkoutData = getCheckoutData(cart)
       stripe.redirectToCheckout(checkoutData)
     }
   }
+
+  if (action.type === 'cart/checkoutSingleItem') {
+    try {
+      stripe = await loadStripe(stripePublicKey)
+    } catch (error) {
+      console.log('error', error)
+    }
+
+    if (action.payload?.sessionId) {
+      return stripe.redirectToCheckout({
+        sessionId: action.payload.sessionId
+      })
+    }
```

Comment:
I feel like `checkoutSingleItem` wouldn't ever need to take a `sessionId` since `redirectToCheckout` already handles that.

Reply:
Makes sense, we'd have to explain in the docs that checkoutSIngleItem is only for client-mode